ring.go

```diff
@@ -73,40 +80,47 @@ func (r *ring) addHostIfMissing(host *HostInfo) (*HostInfo, bool) {
 	if host.invalidConnectAddr() {
 		panic(fmt.Sprintf("invalid host: %v", host))
 	}
-	ip := host.ConnectAddress().String()
+	hostID := host.HostID()
 
 	r.mu.Lock()
 	if r.hosts == nil {
 		r.hosts = make(map[string]*HostInfo)
 	}
+	if r.hostIPToUUID == nil {
+		r.hostIPToUUID = make(map[string]string)
+	}
 
-	existing, ok := r.hosts[ip]
+	existing, ok := r.hosts[hostID]
 	if !ok {
-		r.hosts[ip] = host
+		r.hosts[hostID] = host
+		r.hostIPToUUID[host.ConnectAddress().String()] = hostID
```

Comment:
From what I can tell reading the cassandra source, it looks like the topology events are triggered with the host set to `getNativeaddress` which returns the rpc_address. I think this should be changed to the rpc_address (maybe falling back to the connect address if that's empty?) to prevent any weird issues with connecting through a LB or proxy.

Reply:
Good catch, it was actually a bug in current driver. Nodes are identified between each other and in CQL event by their broadcast_address, which is kept in `broadcast_address` column in `system.local` and by `peer` column in `system.peers`. 
I added new method to HostInfo to extract correct IP and used it in IP to UUID mapping.

Luckily driver was prepared for not finding given host in every event handler so it will just minimize number of ring refreshes.